### ISSUE TITLE
new check: warn about using set with default values

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1933]
#### For new checks and feature suggestions

- [ ] ~shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this~
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I can't check shellcheck.net because it requires JavaScript.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
set +x
echo foo
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

You have used `set +x`, which is the default, did you mean `set -x` instead?

#### Comment

ShellCheck could trace all use of the `set` builtin and determine what the current values are and where the `set` commands have no effect because they either `set` to the default or they `set` to the same value as an earlier `set` command.

**Comment by brother** [comment]
As long as the current script isn't sourced into something else that could have had the default bit flipped...

```
brother ~$ cat /tmp/a.sh 
#!/bin/sh
echo foo
set +x
echo bar
brother ~$ cat /tmp/b.sh 
#!/bin/sh
set -x
echo sourcing
. /tmp/a.sh
echo done
brother ~$ /tmp/b.sh
+ echo sourcing
sourcing
+ . /tmp/a.sh
+ echo foo
foo
+ set +x
bar
done
```
